components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -151,7 +141,14 @@ impl CanvasContext for CanvasRenderingContext2D {
     }
 
     fn get_image_data(&self) -> Option<Vec<u8>> {
-        Some(self.get_rect(Rect::from_size(self.size().cast())))
+        if self.canvas_state.is_paintable() {
+            Some(
+                self.canvas_state
+                    .get_rect(self.size(), Rect::from_size(self.size())),
+            )
+        } else {
+            None
+        }
```

Comment:
Shouldn't we do this as part of get_rect?

Reply:
Fixed as `CanvasState::get_rect() -> Option<Vec<u8>>`